rmf_fleet_adapter/src/rmf_fleet_adapter/agv/EasyFullControl.cpp

```diff
@@ -1055,9 +1055,21 @@ void EasyCommandHandle::follow_new_path(
   for (std::size_t i = 0; i < waypoints.size(); ++i)
   {
     const auto& wp = waypoints[i];
-    if (wp.graph_index().has_value())
+    for (const auto& l : current_location)
     {
-      for (const auto& l : current_location)
+      if (!nav_params->skip_rotation_commands)
+      {
+        // If the waypoint and robot orientations are not aligned with
+        // skip rotation command set to False, we will not consider this as
+        // a connection
+        if (std::abs(wp.position()[2] -
+          l.orientation())*180.0 / M_PI > 1e-2)
+        {
+          continue;
+        }
+      }
+
+      if (wp.graph_index().has_value())
```

Comment:
I just realized you switched the position of:

```
if (wp.graph_index().has_value())
```

and

```
for (const auto& l : current_location)
```

Since `wp` is constant throughout the loop iterations it makes sense to keep it outside of the loop and skip the whole loop if the condition is false, otherwise in cases where `wp = std::nullopt` we would be iterating and doing the `skip_rotation_commands` check just to do nothing in the loop afterwards

Reply:
Ah nevermind just realized the branching is more complex than I thought